### PR TITLE
new OT Stage field `ot_setup_status`

### PR DESCRIPTION
**Pull request by DanielRyanSmith**
Part of #3903.

Creates a new field to denote the status of origin trial creation.

### CODE REVIEW

File: internals/data_types.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ class StageDict(TypedDict):
   ot_is_deprecation_trial: bool
   ot_owner_email: str | None
   ot_require_approvals: bool
+  ot_setup_status: NotRequired[int]
```
Comment (KyleJu) [inline internals/data_types.py:63]: Should this be a required field and backfilled for existing Trials?

**Review by KyleJu — APPROVED** [review]
LGTM with a question

**Comment by KyleJu** [comment]
One more thing - adding a test for changes in converters.py

File: internals/data_types.py (hunk 1 above)
Comment (DanielRyanSmith) [reply internals/data_types.py:63]: We _could_ backfill this here for other origin trial stages, but there shouldn't be a need as long as we're querying specifically for OT stages with `OT_READY_FOR_CREATION` values, meaning that the old OT stages will be ignored.

But if we want to for data consistency purposes, it's safe to assume that any stage with a set `origin_trial_id` has already been created, so we could backfill that value this way.

**Pull request merged**